firedrake/assemble.py

```diff
@@ -20,7 +20,7 @@
 
 
 def assemble(f, tensor=None, bcs=None, form_compiler_parameters=None,
-             inverse=False, nest=None):
+             inverse=False, nest=None, matfree=False, extra_ctx={}):
```

Comment:
I will not get to it today now, go ahead.  If you don't do it, I will go tomorrow.